### PR TITLE
feat[CL]: single migration entry point

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #5306 

## What is the purpose of the change

We currently have two separate endpoints for migrating to a CL pool
1. Migrating with any lock status
2. Migrating with no lock status

We only really need a single entry point, and simply provide a zero value for the lockID if the underlying assets that are being migrated are not locked.

## Testing and Verifying

- Test added to ensure proper routing if lockID is not provided
- Since we are calling a method that already existed in GAMM prior to this PR, that method is extensively tested in the module itself

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/superfluid/keeper/migrate.go [hunk 1]
```diff
@@ -40,6 +41,9 @@ func (k Keeper) RouteLockedBalancerToConcentratedMigration(ctx sdk.Context, send
 		positionId, amount0, amount1, liquidity, joinTime, concentratedLockId, poolIdLeaving, poolIdEntering, err = k.migrateSuperfluidUnbondingBalancerToConcentrated(ctx, sender, lockId, sharesToMigrate, synthLockBeforeMigration.SynthDenom, tokenOutMins)
 	case NonSuperfluid:
 		positionId, amount0, amount1, liquidity, joinTime, concentratedLockId, poolIdLeaving, poolIdEntering, err = k.migrateNonSuperfluidLockBalancerToConcentrated(ctx, sender, lockId, sharesToMigrate, tokenOutMins)
+	case Unlocked:
```
Comment (AlpinYukseloglu) [inline x/superfluid/keeper/migrate.go:44]: Oh wow so the whole second message fits into a single case here?

Comment (czarcas7ic) [reply x/superfluid/keeper/migrate.go:44]: Yep!

File: x/superfluid/keeper/migrate_test.go [hunk 2]
```diff
@@ -616,6 +629,73 @@ func (s *KeeperTestSuite) TestMigrateNonSuperfluidLockBalancerToConcentrated() {
...
+			coinsToMigrate.Amount = coinsToMigrate.Amount.ToDec().Mul(tc.percentOfSharesToMigrate).RoundInt()
+
+			// RouteMigration is called via the migration message router and is always run prior to the migration itself
+			synthLockBeforeMigration, migrationType, err := superfluidKeeper.RouteMigration(ctx, poolJoinAcc, 0, coinsToMigrate)
+			s.Require().NoError(err)
+			s.Require().Equal((lockuptypes.SyntheticLock{}), synthLockBeforeMigration)
+			s.Require().Equal(migrationType, keeper.Unlocked)
+
+			// System under test.
+			positionId, amount0, amount1, _, _, poolIdLeaving, poolIdEntering, err := gammKeeper.MigrateUnlockedPositionFromBalancerToConcentrated(ctx, poolJoinAcc, coinsToMigrate, tc.tokenOutMins)
```
Comment (AlpinYukseloglu) [inline x/superfluid/keeper/migrate_test.go:677]: Should we be checking the liquidity and jointime return values as well?

File: x/superfluid/keeper/migrate_test.go [hunk 3]
```diff
@@ -616,6 +629,73 @@ func (s *KeeperTestSuite) TestMigrateNonSuperfluidLockBalancerToConcentrated() {
...
+			if tc.expectedError != nil {
+				s.Require().Error(err)
```
Comment (AlpinYukseloglu) [inline x/superfluid/keeper/migrate_test.go:679]:
```suggestion
```

File: x/superfluid/keeper/migrate_test.go [hunk 4]
```diff
@@ -616,6 +629,73 @@ func (s *KeeperTestSuite) TestMigrateNonSuperfluidLockBalancerToConcentrated() {
...
+	defaultJoinTime := s.Ctx.BlockTime()
+	type sendTest struct {
+		unlocking                bool
+		percentOfSharesToMigrate sdk.Dec
+		tokenOutMins             sdk.Coins
+		expectedError            error
+	}
+	testCases := map[string]sendTest{
+		"no lock (full shares)": {
+			percentOfSharesToMigrate: sdk.MustNewDecFromStr("1"),
```
Comment (AlpinYukseloglu) [inline x/superfluid/keeper/migrate_test.go:642]: Would be great if we can also add cases for:
1. `percentOfSharesToMigrate ` > 1 (assuming this should error)
2. `percentOfSharesToMigrate` = 0

**Review by AlpinYukseloglu — APPROVED** [review]
Nice cleanup, overall LGTM (left a couple non-blocking comments)

File: x/superfluid/keeper/migrate_test.go (hunk 4 above)
Comment (czarcas7ic) [reply x/superfluid/keeper/migrate_test.go:642]: added extra test cases here https://github.com/osmosis-labs/osmosis/pull/5360/commits/d95a5bae6da27bb73a7208c01870a4cd9bd5cf4e

File: x/superfluid/keeper/migrate_test.go (hunk 2 above)
Comment (czarcas7ic) [reply x/superfluid/keeper/migrate_test.go]: Yeah good call, added this to the other tests too! https://github.com/osmosis-labs/osmosis/pull/5360/commits/c679ac2d85469476775aff1ea815dc3c16fd0c10

**Review by p0mvn — APPROVED** [review]

**Pull request merged**